presto-main/src/main/java/com/facebook/presto/operator/aggregation/histogram/HashUtil.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.histogram;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static it.unimi.dsi.fastutil.HashCommon.arraySize;
+
+public class HashUtil
+{
+    private HashUtil()
+    {
+        throw new AssertionError("static util class only, do not try to instantiate");
```

Comment:
Not necessary

Reply:
no, but again, see bloch, 3rd edition even cites this an extra safety net. guard against reflection. or we can claim java 9 will protect against this (assuming we setup proper modules)

else, what's the harm? it's easy to use reflection to get around a private constructor

Reply:
leaving for now

Reply:
I can see where that might be reasonable for a generic library, but this is not the case. We don’t use reflection that way in presto, so this will never happen.

Reply:
ok i'll remove it

that said, what's the harm? it's an extra guard. i suppose you're expecting review from catching any trying to do that